File: Week1/PlusMinus/src/main/java/MyCode.java
package main.java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.List;
import java.util.stream.Stream;

import static java.util.stream.Collectors.toList;

public class MyCode {

    public static void plusMinus(List<Integer> arr){
        float positives = ((float) arr.stream().filter(v-> v >= 0 && v <= 100 ).count());
        float negatives = ((float) arr.stream().filter(v-> v <= 0 && v >= -100).count());
        float zeros = ((float) arr.stream().filter(v-> v == 0).count());

        System.out.printf("%.6f",positives);
        System.out.printf("%.6f",negatives);
        System.out.printf("%.6f",zeros);

        System.out.printf("%.6f",positives/arr.size());
        System.out.printf("%.6f",negatives/arr.size());
        System.out.printf("%.6f",zeros/arr.size());
    }
}

File: Week1/PlusMinus/src/main/kotlin/MyCode.kt
package main.kotlin

fun plusMinus(arr:Array<Int>){
    var zeros: Float = 0.0f
    var positives: Float = 0.0f
    var negatives: Float = 0.0f

    for (i in arr){
        when(i){
            0->zeros++
            in -1000 .. -1 -> negatives++
            in 1 .. 1000 ->positives++

        }
    }
    println(String.format("%.6f",positives/arr.size))
    println(String.format("%.6f",negatives/arr.size))
    println(String.format("%.6f",zeros/arr.size))
}
fun main(args: Array<String>){
    val n = readLine()!!.trim().toInt()

    val arr = readLine()!!.trimEnd().split(" ").map{ it.toInt() }.toTypedArray()

    plusMinus(arr)
}